src/libraries/System.Xml.ReaderWriter/ref/System.Xml.ReaderWriter.cs

```diff
@@ -117,17 +117,18 @@ public partial class XmlAttribute : System.Xml.XmlNode
         public override string NamespaceURI { get { throw null; } }
         public override System.Xml.XmlNodeType NodeType { get { throw null; } }
         public override System.Xml.XmlDocument OwnerDocument { get { throw null; } }
-        public virtual System.Xml.XmlElement OwnerElement { get { throw null; } }
-        public override System.Xml.XmlNode ParentNode { get { throw null; } }
+        public virtual System.Xml.XmlElement? OwnerElement { get { throw null; } }
+        public override System.Xml.XmlNode? ParentNode { get { throw null; } }
         public override string Prefix { get { throw null; } set { } }
         public override System.Xml.Schema.IXmlSchemaInfo SchemaInfo { get { throw null; } }
         public virtual bool Specified { get { throw null; } }
+        [System.Diagnostics.CodeAnalysis.AllowNullAttribute]
         public override string Value { get { throw null; } set { } }
```

Comment:
Shouldn't `InnerText` also have `[AllowNull]`?
https://github.com/dotnet/runtime/blob/53f4c08b5442733e81e82d85646f0e4cacf7bae8/src/libraries/System.Private.Xml/src/System/Xml/Dom/XmlAttribute.cs#L109-L114

Reply:
I think I've tried many combinations of annotations of Value, InnerText and Data (IIRC) and they are a giant mess and you can annotate them randomly and will be as good as now - the implementations are inconsistent with each other. We'll likely reiterate on them in 6.0. I think current state was producing least errors but it's still buggy